user_tools/src/spark_rapids_tools/tools/additional_heuristics.py

```diff
@@ -68,18 +76,36 @@ def _apply_heuristics(self, app_ids: list) -> pd.DataFrame:
         else:
             for app_id in app_ids:
                 app_id_path = os.path.join(metrics_path, app_id)
-                try:
-                    # Apply heuristics and determine if the application should be skipped.
-                    # Note: `should_skip` flag can be a combination of multiple heuristic checks.
-                    should_skip, reason = self.heuristics_based_on_spills(app_id_path)
-                except Exception as e:  # pylint: disable=broad-except
-                    should_skip = False
-                    reason = f'Cannot apply heuristics for qualification. Reason - {type(e).__name__}:{e}'
-                    self.logger.error(reason)
-                result_arr.append([app_id, should_skip, reason])
+                # Apply a list of heuristics and determine if the application should be skipped.
+                should_skip_overall = False
+                reasons = []
+                for heuristic_func in self._get_all_heuristics_functions():
+                    try:
+                        should_skip, reason = heuristic_func(app_id_path)
+                    except Exception as e:  # pylint: disable=broad-except
+                        should_skip = False
+                        reason = f' Cannot apply heuristics for qualification. Reason - {type(e).__name__}:{e}.'
+                        self.logger.error(reason)
+                    should_skip_overall = should_skip_overall or should_skip
+                    reasons.append(reason)
+                result_arr.append([app_id, should_skip_overall, ' '.join(reasons)])
 
         return pd.DataFrame(result_arr, columns=self.props.get_value('resultCols'))
 
+    def heuristics_based_on_total_core_seconds(self, app_id_path: str) -> (bool, str):
+        """
+        Apply heuristics based on total core seconds to determine if the app can be accelerated on GPU.
+        """
+        # Load app total core seconds from qualidication summary dataframe
+        app_id = app_id_path.split('/')[-1]
```

Comment:
nit: Can we use python library to extract the dir name?

Reply:
fixed, thanks!